ChangeLog.d/fix-driver-schema-check.txt

```diff
@@ -0,0 +1,3 @@
+Bugfix
+   * Fix type definition's default value in driver_*_schema.json. Also removes
+     warning, if the right string definition is used.
```

Comment:
The audience of changelog entries is users. The entry currently describes the content of the patch, but not what it means from a user's perspective. Something like this:

> Fix a warning about an invalid JSON schema when running generate_driver_wrappers.py.

Although I can't reproduce the warning here, even though running `scripts/generate_driver_wrappers.py` with default options should load our bundled drivers (test and p256-m) which do have a `"type"` property that is incorrectly described in the schema.

Note that I agree that your patch is correct. I'm just wondering why you're getting an error due to the current invalid schema and I'm not.

Reply:
I'm not entirely sure about how `generate_driver_wrappers.py` works, and I wasn't talking about seeing a warning at compile time.

I meant, when I'm editing `my_driver_tansparent/opaque_driver.json` file, and using the cshema in it for help, and autocorrect with LSP (specifying `"$schema": "https://github.com/Mbed-TLS/mbedtls/blob/development/ scripts/data_files/driver_jsons/driver_transparent_schema.json"` at the begining) with [vscode-langservers-extracted](https://github.com/hrsh7th/vscode-langservers-extracted), it will automatically set the `"type":` as `["transparent"]`, but show an error about the type being string not an array. After deleting the brackets and setting `"type": "transparent"` it will show a warning about type should be `["transparent"]`.

Reply:
Ah, I see, thanks. The information I was missing was that some other tool complained. Our Python script just calls Python's `jsonschema` package to load the schema, and I guess that package doesn't validate the schema itself. I've filed a [backlog issue about checking the validity of the schemas](https://github.com/Mbed-TLS/mbedtls/issues/9678).

So can you please change the changelog entry to something that just says the schema was invalid? E.g.

> Fix invalid JSON schemas for driver descriptions used by generate_driver_wrappers.py.

Reply:
Done